x-pack/plugins/security_solution/server/lib/detection_engine/signals/send_telemetry_events.ts

```diff
@@ -31,13 +45,20 @@ export function sendAlertTelemetryEvents(
   logger: Logger,
   eventsTelemetry: ITelemetryEventsSender | undefined,
   filteredEvents: SignalSearchResponse,
+  createdEvents: SignalSource[],
   buildRuleMessage: BuildRuleMessage
 ) {
   if (eventsTelemetry === undefined) {
     return;
   }
+  // Create map of ancenstor_id -> alert_id
+  let signalIdMap = new Map<string, string>();
```

Comment:
We build a map of `<endpoint_alert_id>, `<signal_id>` so that we can look it up when we send the payload

Reply:
This code isn't pragmatic. 

```ts
createdEvents.map(function (obj: SignalSource) {
  signalIdMap = signalIdMap.set(String(obj['kibana.alert.original_event.id']), String(obj._id));
  return null;
});
```

Yields an array of nulls where the length is the number of events. You are also side-effecting unnecessarily and not checking if the `ancestor_id` exists. A safer and more functional way of writing this code would be:

```ts
type AncestorId = string;
type AlertId = string;

const signalIdMap = createdEvents.reduce((signalMap, obj) => {
  const ancestorId = String(obj['kibana.alert.original_event.id']);
  const alertId = String(obj._id);
    
  if (ancestorId !== null && ancestorId !== undefined) {
    signalMap = signalIdMap.set(ancestorId, alertId);
  }
    
  return signalMap;
}, new Map<AncestorId, AlertId>())
```

I think I would also like to see a unit test around this function - one where the ancestor id exists and another when it doesn't.